uvicorn/protocols/http/httptools_impl.py

```diff
@@ -38,38 +43,38 @@ def _get_status_line(status_code):
 
 
 class FlowControl:
-    def __init__(self, transport):
+    def __init__(self, transport: asyncio.Transport) -> None:
         self._transport = transport
         self.read_paused = False
         self.write_paused = False
         self._is_writable_event = asyncio.Event()
         self._is_writable_event.set()
 
-    async def drain(self):
+    async def drain(self) -> None:
         await self._is_writable_event.wait()
 
-    def pause_reading(self):
+    def pause_reading(self) -> None:
         if not self.read_paused:
             self.read_paused = True
             self._transport.pause_reading()
 
-    def resume_reading(self):
+    def resume_reading(self) -> None:
         if self.read_paused:
             self.read_paused = False
             self._transport.resume_reading()
 
-    def pause_writing(self):
+    def pause_writing(self) -> None:
         if not self.write_paused:
             self.write_paused = True
             self._is_writable_event.clear()
 
-    def resume_writing(self):
+    def resume_writing(self) -> None:
         if self.write_paused:
             self.write_paused = False
             self._is_writable_event.set()
 
 
-async def service_unavailable(scope, receive, send):
+async def service_unavailable(scope: Scope, receive: Callable, send: Callable) -> None:
```

Comment:
The idea is to follow `asgiref`. One of my PRs already have the types for `receive` and `send`.